src/Behat/Behat/Context/Snippet/Generator/ContextSnippetGenerator.php

```diff
@@ -122,7 +123,7 @@ public function generateSnippet(Environment $environment, StepNode $step): Snipp
      */
     private function getMethodName(string $contextClass, string $canonicalText, string $pattern): string
     {
-        $methodName = $this->deduceMethodName($canonicalText);
+        $methodName = SimpleStepMethodNameGenerator::cleanupMethodName($canonicalText);
```

Comment:
We need to keep this here for now in case anyone has a custom PatternPolicy that is not implementing our logic. We can remove it at the next major.